src/hotspot/share/cds/archiveBuilder.cpp

```diff
@@ -461,8 +454,16 @@ bool ArchiveBuilder::gather_one_source_obj(MetaspaceClosure::Ref* enclosing_ref,
   }
 }
 
-void ArchiveBuilder::remember_embedded_pointer_in_copied_obj(MetaspaceClosure::Ref* enclosing_ref,
-                                                             MetaspaceClosure::Ref* ref) {
+void ArchiveBuilder::remember_embedded_pointer_in_gathered_obj(address src_obj, address* field_addr) {
+  // field_addr must be a valid field inside src_obj.
```

Comment:
Is there a way to check this with an assert?

Reply:
The assert is done inside `ArchiveBuilder::SourceObjList::remember_embedded_pointer()`, which knows the size of `src_obj`:

```
assert(field_offset_in_bytes >= 0, "must be");
assert(field_offset_in_bytes + intx(sizeof(intptr_t)) <= intx(src_obj_size), "must be");
```